File: 프로그래머스/연속된_부분수열의_합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        int right =0;
        int left =0;
        int sum =0;
        int minLength = Integer.MAX_VALUE;
        while(right < sequence.length){
            while(sum < k && right < sequence.length){
                sum += sequence[right++];
            }
            while(sum > k && left < sequence.length){
                sum -= sequence[left++];
            }
            if(sum == k){
                int pointerLength = right - left;
                sum -= sequence[left++];
                if(minLength > pointerLength){
                    minLength = pointerLength;
                    answer[0] = left;
                    answer[1] = right;
                }
            }
        }
        answer[0]--;
        answer[1]--;
        return answer;
    }
}

File: 프로그래머스/연속된_부분_수열의_합.js
function solution(sequence, k) {
    var minLength = Infinity;
    var answer = [];
    var left = 0;
    var right = 0;
    var sum = sequence[0];
    while(right < sequence.length && left <= right){
        if(sum < k){
            right++;
            if(right < sequence.length) sum += sequence[right];
        }
        if(sum > k){
            sum -= sequence[left++];
        }
        if(sum == k){
            if(minLength > (right - left)){
                answer = [left,right];
                minLength = right - left;
            }
            sum -= sequence[left++];
        }
    }
    
    return answer;
}
